babel/numbers.py

```diff
@@ -882,8 +882,18 @@ def parse_number(string: str, locale: Locale | str | None = LC_NUMERIC) -> int:
     :return: the parsed number
     :raise `NumberFormatError`: if the string can not be converted to a number
     """
+    group_symbol = get_group_symbol(locale)
+
+    if (
+        re.match(r"\s", group_symbol) and  # if the grouping symbol is a kind of space,
+        group_symbol not in string and  # and the string to be parsed does not contain it,
+        re.search(r"\s", string)  # but it does contain any other kind of space instead,
+    ):
+        # ... it's reasonable to assume it is taking the place of the grouping symbol.
+        string = re.sub(r"\s", group_symbol, string)
```

Comment:
I'm a little worried about the possible performance issues here. Could 

* this be reordered so the cheapest check is first (`group_symbol not in string and ...`)
* `re.match(r"\s", group_symbol)` moved to an `lru_cache`d helper function – or if `group_symbol.isspace()` is enough, then let's do that instead?

Reply:
Sorry for the delay, I kind of forgot about this PR.

It looks like`group_symbol.isspace()` should work the same as `re.match(r"\s", group_symbol)`, but I'm thinking they both might be too wide and match things like new lines. Maybe it’s better to be more specific and only match `" "`, `"\xa0"` and `"\u202f"`.

Reply:
I think using `.isspace()` might be better than hard-coding the set of spaces. I think we're already missing some space-like characters from that set anyway and I don't think `group_symbol` will ever be set to a newline...